### PR TITLE
check masq-profiles and data hyperdb at initialisation

### PR DESCRIPTION
**Pull request by leventdem**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "description": "",
   "module": "src/index.js",
   "scripts": {
-    "test": "eslint src/ test/ && jest",
+    "test": "jest",
```
Comment (G-Ray) [inline package.json]: eslint step should not be deleted

File: src/index.js [hunk 2]
```diff
@@ -7,6 +7,21 @@ const pump = require('pump')
 
 const HUB_URL = 'localhost:8080'
 
+const isCreated = (dbName) => {
```
Comment (G-Ray) [inline src/index.js]: seems good :+1:

File: src/index.js [hunk 3]
```diff
@@ -7,6 +7,21 @@ const pump = require('pump')
...
+    let req = indexedDB.open(dbName)
+    let existed = true
+    req.onsuccess = () => {
+      req.result.close()
+      if (!existed) { indexedDB.deleteDatabase(dbName) }
+      resolve(existed)
+    }
+    req.onupgradeneeded = () => {
+      existed = false
+    }
```
Comment (G-Ray) [inline src/index.js]: maybe reject `onerror`

File: src/index.js [hunk 4]
```diff
@@ -265,9 +280,24 @@ class Masq {
   }
 
   /** open and sync existing databases */
-  _openAndSyncDatabases () {
-    // const db = hyperdb(rai('masq-profiles'), { valueEncoding: 'json' })
-    // this.dbs.profiles = db
+  async _openAndSyncDatabases () {
+    if (isCreated('masq-profiles')) {
```
Comment (G-Ray) [inline src/index.js]: missing await before isCreated ?

File: src/index.js [hunk 5]
```diff
@@ -265,9 +280,24 @@ class Masq {
...
+      const db = hyperdb(rai('masq-profiles'), { valueEncoding: 'json' })
+      await dbReady(db)
+      this.dbs.profiles = db
+      this._startReplication(db, 'masq-profiles')
+    }
+    let profiles = await this.getProfiles()
+
+    for (let index = 0; index < profiles.length; index++) {
+      let id = profiles[index].id
+      if (isCreated(id)) {
```
Comment (G-Ray) [inline src/index.js]: await ?

File: test/masq.test.js [hunk 6]
```diff
@@ -214,10 +210,21 @@ test('should exchange key and authorize local key if challenge matches', async (
   masq.exchangeDataHyperdbKeys(appName)
 })
 
+/**
+ * Return when hyperDb instance is ready
+ * @param {Object} db - The hyperDb instance
+ */
+const dbReady = (db) => {
```
Comment (G-Ray) [inline test/masq.test.js]: Please move this util function at the top of the test file

File: src/index.js (hunk 2 above)
Comment (sujeebant) [reply src/index.js]: maybe rename the function to dbExists ?

File: package.json (hunk 1 above)
Comment (sujeebant) [reply package.json]: changed

File: src/index.js (hunk 2 above)
Comment (sujeebant) [reply src/index.js]: done

File: src/index.js (hunk 4 above)
Comment (sujeebant) [reply src/index.js]: added

File: src/index.js (hunk 5 above)
Comment (sujeebant) [reply src/index.js]: added

File: test/masq.test.js (hunk 6 above)
Comment (sujeebant) [reply test/masq.test.js]: removed as it was not used

File: src/index.js (hunk 3 above)
Comment (sujeebant) [reply src/index.js]: added

File: src/index.js [hunk 7]
```diff
@@ -7,6 +7,8 @@ const pump = require('pump')
 
 const HUB_URL = 'localhost:8080'
 
+const dbExists = require('./indexedDBUtils').dbExists
```
Comment (G-Ray) [inline src/index.js]: Please move just before HUB_URL

Comment (G-Ray) [reply src/index.js]: Follow the rule:
At the very top: node_modules imports
one line after: relative imports

File: src/index.js [hunk 8]
```diff
@@ -265,9 +267,26 @@ class Masq {
...
-    // this.dbs.profiles = db
+  async _openAndSyncDatabases () {
+    if (!(await dbExists('masq-profiles'))) {
+      return
+    }
+    const db = hyperdb(rai('masq-profiles'), { valueEncoding: 'json' })
+    await dbReady(db)
+    this.dbs.profiles = db
+    this._startReplication(db, 'masq-profiles')
+    let profiles = await this.getProfiles()
```
Comment (G-Ray) [inline src/index.js]: profiles can be defined with `const`

**Pull request merged**